### DISCUSSION TITLE
Has permission to access a route

### DISCUSSION THREAD

**Discussion by frangambin1** [adonisjs/core#3236] [Help]
Hello,

I am currently a Drupal developer and a week ago I found your framework and I wanted to congratulate you on your enormous work.

Thanks to this framework I can dock or make a structure similar to Drupal, developing by modules.

I have two small doubts, one of them resolved since, since the development of internationalization is pending in version 5 of Adonis JS. (Seen on the trello board)

The question is ... how can I check if a user has access to a route.

I describe the scenario:

I have a route where I specify the assigned roles, I attach a screenshot.

<img width="546" alt="Captura de pantalla 2021-10-11 a las 12 35 35" src="https://user-images.githubusercontent.com/3390025/136779778-d1f4318c-2644-45b4-ae88-611749d9c1e4.png">

In the .edge template I generate a path that points to that address (example.view) but I only want to show the link in html format if the authenticated user has access permission.

I hope I have explained myself well.

Many thanks.

**Comment by feniljariwala82** [comment]
As per my knowledge, you don't need to handle permissions here. You can simply apply a middleware, and there you can define your restriction logic. I hope this helps.

**Reply by frangambin1** [reply]
Thank you very much for your reply.

Adonis JS has the possibility of using middleware from the routes file, in my opinion it is cleaner and more viable during development.

The idea or goal is to use a helper in the edge templating engine to find out if an authenticated user has access to a particular path.

**Reply by arthur-er** [reply]
You can use the bouncer package with the @can directives (https://docs.adonisjs.com/guides/authorization#usage-inside-the-edge-templates)
